.github/workflows/unit-test.yml

```diff
@@ -5,7 +5,34 @@ on:
   pull_request:
 
 jobs:
+  build:
```

Comment:
note for reviewer: same here for unit tests. I guess a single compile step can be done for both workflows but I think this will add complexity to them. This way both workflows are independent and still we reduce the resource usage